applications/webhooks/src/webhooks_shared_listener.erl

```diff
@@ -295,14 +295,14 @@ handle_info(_Info, State) ->
 %% @end
 %%--------------------------------------------------------------------
 -spec handle_event(kz_json:object(), kz_proplist()) -> gen_listener:handle_event_return().
-handle_event(_JObj, _State) ->
+handle_event(JObj, _State) ->
     WHZones = lists:sort([kz_util:to_binary(Z)
                           || Z <- kz_nodes:whapp_zones(?APP_NAME)
                          ]),
-    Options = [{<<"should_handle_federated">>
-               ,kz_util:to_binary(kz_nodes:local_zone()) == hd(WHZones)
-               }
-              ],
+    ShouldHandle = kz_api:is_federated_event(JObj)
+        andalso not lists:member(kz_api:event_zone(JObj), WHZones)
+        andalso kz_util:to_binary(kz_nodes:local_zone()) == hd(WHZones),
+    Options = [{<<"should_handle_federated">>, ShouldHandle}],
```

Comment:
@k-anderson we're deciding if we should take a federated message and do the job if there is no node in that zone running webhooks. i would very much prefer to actually remove federation from webhooks listener and have it deal only with events within the zone in a shared queue with a limited persistence to allow restarts.